src/jira/index.ts

```diff
@@ -0,0 +1,39 @@
+import JiraApi = require("jira-client");
+import * as TE from "fp-ts/lib/TaskEither";
+import * as A from "fp-ts/lib/ReadonlyArray";
+import { toError } from "fp-ts/lib/Either";
+import { errorsToReadableMessages } from "@pagopa/ts-commons/lib/reporters";
+import { pipe } from "fp-ts/lib/function";
+import { JiraIssueResponse } from "./types";
+
+export const jiraHostName = "pagopa.atlassian.net";
+const jiraApi = new JiraApi({
+  host: jiraHostName,
+  username: process.env.JIRA_USERNAME,
+  password: process.env.JIRA_PASSWORD,
+});
+
+export const getJiraIssue = (
+  id: string
+): TE.TaskEither<Error, JiraIssueResponse> =>
+  pipe(
+    TE.tryCatch(() => jiraApi.getIssue(id), toError),
+    TE.chain((response) =>
+      pipe(
+        TE.fromEither(JiraIssueResponse.decode(response)),
+        TE.mapLeft(
+          (errs) =>
+            new Error(
+              `Cannot decode Response|${errorsToReadableMessages(errs).join(
+                "/"
+              )}`
+            )
+        )
+      )
```

Comment:
I suggest to extract nested pipes into new methods (so you can avoid them)

Reply:
Here you probably can avoid the nested pipe using `TE.chainEitherW`

Reply:
TE.chainEitherW(JiraIssueResponse.decode) use `Errors` that is an array of `ValidationError` instead of `Error`. For now I have divided the pipes